src/server/accounting.c

```diff
@@ -2259,3 +2259,86 @@ void log_alter_records_for_attrs(job *pjob, svrattrl *plist) {
 	if (entire_record[0] != '\0')
 		account_record(PBS_ACCT_ALTER, pjob, entire_record);
 }
+
+
+/**
+ * @brief
+ * Common function to log a suspend/resume record 
+ * for suspend/resume job events respectively.
+ * 
+ * @param[in] pjob - job to log records for.
+ * @param[in] user - User who requested the signal suspend/resume 
+ * @param[in] host - Hostname where the signal was triggered
+ * @param[in] signal_type - flag indicating the type of event suspend/resume
+ * 
+ * @returns void 
+ */
+void
+log_suspend_resume_record(job *pjob, char *user, char *host, char *signal_type)
+{
+	char *log_record = NULL;
+	char *tmp;
+	struct svrattrl *patlist = NULL;
+	char *resc_used;
+	int resc_used_size = 0;
+	pbs_list_head temp_head;
+	int acct_type = PBS_ACCT_RESUME;
+	int rc;
+
+	rc = pbs_asprintf(&log_record, "requestor=%s@%s", user, host);
+
+	if (strcmp(signal_type, SIG_SUSPEND)==0 || strcmp(signal_type, SIG_ADMIN_SUSPEND) == 0) {
+		acct_type = PBS_ACCT_SUSPEND;
+		
+		if (pjob->ji_wattr[JOB_ATR_resc_released].at_flags & ATR_VFLAG_SET) {
+			tmp = log_record;
```

Comment:
Hi @bhroam, I am using `tmp` basically to copy the initial string to the newly allocated memory by `pbs_asprintf` into `log_record`. If I free `log_record` here, I will lose the previously stored string. I hope my understanding is correct.

Reply:
Yes, you will lose the previously stored value, but you do anyway when you free(tmp) 2 lines down.  There isn't really isn't any difference between freeing it here and freeing it 2 lines down because you don't do anything with the value in between.

Reply:
Oh, I see, you use tmp IN the pbs_asprintf() call.  That is dangerous because you are now setting and using the same value in the same call.  You don't know what pbs_asprintf() is going to do.  You are probably better off doing 2 calls to pbs_strcat().